tool/site/client/src/services/test-packs/index.ts

```diff
@@ -0,0 +1,257 @@
+      patientEntry,
+      practitionerEntry,
+      practitionerRoleEntry,
+      createCommunicationRequest(patientEntry)
+    ]
+  }
+  createMedicationRequests(
+    prescriptionRows,
+    repeatsIssued,
+    maxRepeatsAllowed
+  ).forEach(medicationRequest =>
+    fhirPrescription.entry.push(medicationRequest)
+  )
+  createPlaceResources(prescriptionType, fhirPrescription)
+  return JSON.stringify(fhirPrescription)
+}
+
+export function getPrescriptionTreatmentType(row: PrescriptionRow, setLoadPageErrors?: Dispatch<SetStateAction<any>>): TreatmentType {
+  const code = row.prescriptionTreatmentTypeCode
+  if (!validFhirPrescriptionTypes.includes(code)) {
+    // eslint-disable-next-line max-len
+    const treatmentTypeInvalidError = `Treatment Type column contained an invalid value. 'Prescription Type' must be one of: ${validFhirPrescriptionTypes.join(", ")}`
+    if (setLoadPageErrors) {
+      setLoadPageErrors({details: [treatmentTypeInvalidError]})
+    }
+    throw new Error(treatmentTypeInvalidError)
+  }
+  switch (code) {
+    case "acute":
+      return "acute"
+    case "repeat-prescribing":
+      return "continuous"
+    case "repeat-dispensing":
+      return "continuous-repeat-dispensing"
+  }
+}
+
+function getPrescriptionTypeFromRow(prescriptionRows: Array<PrescriptionRow>): PrescriptionType {
+  const row = prescriptionRows[0]
+  const prescriptionTypeCode = row.prescriptionTypeCode
+  return getPrescriptionType(prescriptionTypeCode)
+}
+
+function updateNominatedPharmacy(bundle: fhir.Bundle, odsCode: string): void {
+  if (!odsCode) {
+    return
+  }
+  getMessageHeaderResources(bundle).forEach(messageHeader => {
+    messageHeader.destination.forEach(destination => {
+      destination.receiver.identifier.value = odsCode
+    })
+  })
+  getMedicationRequestResources(bundle).forEach(function (medicationRequest) {
+    medicationRequest.dispenseRequest.performer = {
+      identifier: {
+        system: "https://fhir.nhs.uk/Id/ods-organization-code",
+        value: odsCode
+      }
+    }
+  })
+}
+
+function getNominatedPharmacyOdsCode(nominatedPharmacies: Array<string>, prescriptionRow: PrescriptionRow) {
+  if (!prescriptionRow.testId) {
+    return null
+  }
+  const testNumber = parseInt(prescriptionRow.testId)
+  return nominatedPharmacies[testNumber - 1]
+}
+
+export type TreatmentType = "acute" | "continuous" | "continuous-repeat-dispensing"
+
+export type PrescriptionType = "prescribing-cost-centre-0101" | "prescribing-cost-centre-non-0101" | "trust-site-code"
+
+export function getPrescriptionType(prescriberType: string): PrescriptionType {
+  if (prescriberType.startsWith("10")) {
+    return "trust-site-code"
+  }
+
+  switch (prescriberType) {
+    case "0101":
+      return "prescribing-cost-centre-0101"
+    case "0104":
+    case "0105":
+    case "0108":
+    case "0125":
+      return "prescribing-cost-centre-non-0101"
+    default:
+      throw new Error("Prescription type not handled")
+  }
+}
```

Comment:
I've been through this with as much attention as I can manage, looks good but it is possible I've missed something